### PR TITLE
realigning table config for market access

### PR DESCRIPTION
**Pull request by sekharpanja**
### Description of change
Market access pipeline table config need few changes as API response has changed significantly.

Here is the concerning part of the new API response:

```    "economic_assessments": [
      {
        "id": 1,
        "approved": true,
        "archived": false,
        "archived_by": null,
        "archived_on": null,
        "archived_reason": "",
        "automated_analysis_data": null,
        "barrier_id": "123",
        "created_by": {
          "id": 223,
          "name": "Foo bar"
        },
        "created_on": "2019-08-29T08:47:08.487872Z",
        "documents": [
          {
            "id": "123",
            "name": "B-19-ABC (services tax) summary.xlsx",
            "size": 72903,
            "status": "virus_scanned"
          }
        ],
        "economic_impact_assessments": [],
        "explanation": "blah",
        "export_value": 100,
        "import_market_size": 100,
        "modified_on": "2020-11-30T11:47:47.981305Z",
        "modified_by": {
          "id": 223,
          "name": "Foo bar"
        },
        "rating": {
          "code": "HIGH",
          "name": "High"
        },
        "ready_for_approval": true,
        "reviewed_by": {
          "id": 223,
          "name": "Foo bar"
        },
        "reviewed_on": "2019-08-29T08:47:08.487872Z",
        "value_to_economy": null
      }
    ],```

### Checklist

* [ ] Have tests been added to cover any changes?
* [ ] Has the README been updated (if needed)?

### CODE REVIEW

File: dataflow/dags/market_access_pipelines.py [hunk 1]
```diff
@@ -57,33 +57,52 @@ class MarketAccessTradeBarriersPipeline(_PipelineDAG):
...
             (
-                ('economic_assessment', 'economic_impact_assessments'),
+                'economic_assessments',
                 TableConfig(
-                    table_name='market_access_economic_impact_assessments',
+                    table_name='market_access_economic_assessments',
+                    transforms=[
+                        lambda record, table_config, contexts: {
+                            **record,
+                            'trade_barrier_id': contexts[0]['id'],
```
Comment (abbas123456) [inline dataflow/dags/market_access_pipelines.py:66]: Each economic assessment object already has a field called `barrier_id` so you dont need any transforms here

File: dataflow/dags/market_access_pipelines.py [hunk 2]
```diff
@@ -57,33 +57,52 @@ class MarketAccessTradeBarriersPipeline(_PipelineDAG):
...
+                        ('export_value', sa.Column('export_value', sa.BigInteger),),
+                        ('trade_barrier_id', sa.Column('trade_barrier_id', sa.Text)),
+                        (
+                            ('economic_assessments', 'economic_impact_assessments'),
+                            TableConfig(
+                                table_name='market_access_economic_impact_assessments',
+                                transforms=[
+                                    lambda record, table_config, contexts: {
+                                        **record,
+                                        'trade_barrier_id': contexts[0]['id'],
```
Comment (abbas123456) [inline dataflow/dags/market_access_pipelines.py:88]: You need to link each `economic_impact_assessments` object to its corresponding `economic_assessments` object, not the top level trade barrier

**Review by abbas123456 — CHANGES_REQUESTED** [review]

File: dataflow/dags/market_access_pipelines.py [hunk 3]
```diff
@@ -57,33 +57,52 @@ class MarketAccessTradeBarriersPipeline(_PipelineDAG):
...
+                        ('export_value', sa.Column('export_value', sa.BigInteger),),
+                        ('trade_barrier_id', sa.Column('trade_barrier_id', sa.Text)),
+                        (
+                            ('economic_assessments', 'economic_impact_assessments'),
+                            TableConfig(
+                                table_name='market_access_economic_assessment_impact_assessments',
+                                transforms=[
+                                    lambda record, table_config, contexts: {
+                                        **record,
+                                        'assessment_id': contexts[0]['id'],
```
Comment (abbas123456) [inline dataflow/dags/market_access_pipelines.py:88]: Similar to the above comment, the EconomicImpactAssessment object already has a link to its EconomicAssesesment object via a field called `economic_assessment_id`. Therefore you don't need this transform. You need to rename the `assessment_id` field in the field_mapping list to `economic_assessment_id`

File: dataflow/dags/market_access_pipelines.py [hunk 4]
```diff
@@ -57,33 +57,52 @@ class MarketAccessTradeBarriersPipeline(_PipelineDAG):
...
-                            sa.Column('economic_assessment_id', UUID, index=True),
+                            'value_to_economy',
+                            sa.Column('value_to_economy', sa.BigInteger),
+                        ),
+                        (
+                            'import_market_size',
+                            sa.Column('import_market_size', sa.BigInteger),
+                        ),
+                        ('export_value', sa.Column('export_value', sa.BigInteger),),
+                        ('trade_barrier_id', sa.Column('trade_barrier_id', sa.Text)),
```
Comment (abbas123456) [inline dataflow/dags/market_access_pipelines.py:80]: This needs to be renamed to `barrier_id` and the related transform needs to be removed as stated above

**Review by samuelhwilliams — APPROVED** [review]

File: dataflow/dags/market_access_pipelines.py [hunk 5]
```diff
@@ -57,33 +57,40 @@ class MarketAccessTradeBarriersPipeline(_PipelineDAG):
...
+                            ('economic_assessments', 'economic_impact_assessments'),
+                            TableConfig(
+                                table_name='market_access_economic_assessment_impact_assessments',
+                                field_mapping=[
+                                    ('id', sa.Column('id', UUID, index=True)),
+                                    (('impact', 'name'), sa.Column('impact', sa.Text)),
+                                    ('archived', sa.Column('archived', sa.Boolean)),
+                                    ('explanation', sa.Column('explanation', sa.Text),),
+                                    (
+                                        'assessment_id',
```
Comment (abbas123456) [inline dataflow/dags/market_access_pipelines.py:85]: This needs to be `economic_assessment_id`

File: dataflow/dags/market_access_pipelines.py [hunk 6]
```diff
@@ -57,33 +57,40 @@ class MarketAccessTradeBarriersPipeline(_PipelineDAG):
...
+                            sa.Column('value_to_economy', sa.BigInteger),
+                        ),
+                        (
+                            'import_market_size',
+                            sa.Column('import_market_size', sa.BigInteger),
+                        ),
+                        ('export_value', sa.Column('export_value', sa.BigInteger),),
+                        ('barrier_id', sa.Column('barrier_id', sa.Text)),
+                        (
+                            ('economic_assessments', 'economic_impact_assessments'),
```
Comment (abbas123456) [inline dataflow/dags/market_access_pipelines.py:76]: This shouldn't be a nested tuple, should just be `economic_impact_assessments`

**Review by abbas123456 — APPROVED** [review]

**Pull request merged**